dotnet/src/Skills/Skills.OpenAPI/Authentication/README.md

```diff
@@ -0,0 +1,66 @@
+# Authentication for the OpenAPI Skill
+
+The Semantic Kernel OpenAPI Skill enables developers to take any REST API that follows the OpenAPI specification and import it as a skill to the Semantic Kernel. However, the Kernel needs to be able to authenticate outgoing requests per the requirements of the target API. This document outlines the authentication model for the OpenAPI skill as well as the reference implementations provided by the Semantic Kernel.
+
+## The `AuthenticateRequestAsyncCallback` delegate
+
+[`AuthenticateRequestAsyncCallback`](../../../SemanticKernel.Abstractions/Connectors/WebApi/Rest/AuthenticateRequestAsyncCallback.cs) is a delegate type that serves as a callback function for adding authentication information to HTTP requests sent by the OpenAPI skill.
+
+```csharp
+public delegate Task AuthenticateRequestAsyncCallback(HttpRequestMessage request);
+```
+
+Developers may optionally provide an implementation of this delegate when importing an OpenAPI skill to the Kernel. The delegate is then passed through to the `RestApiOperationRunner`, which is responsible for building the HTTP payload and sending the request for each REST API operation. Before the API request is sent, the delegate is executed with the HTTP request message as the parameter, allowing the request message to be updated with any necessary authentication information.
+
+This pattern was designed to be flexible enough to support a wide variety of authentication frameworks. Developers can provide the delegate function directly or define a class or interface that exposes one or more implementations. They have the option of writing their own custom implementation or using one of the Semantic Kernel's reference authentication providers as a starting point.
+
+## Reference Authentication Providers
+
+### [`BasicAuthenticationProvider`](./BasicAuthenticationProvider.cs)
+This class implements the HTTP "basic" authentication scheme. The constructor accepts a `Func` which defines how to retrieve the user's credentials. When the `AuthenticateRequestAsync` method is called, it retrieves the credentials, encodes them as a UTF-8 encoded Base64 string, and adds them to the `HttpRequestMessage`'s authorization header.
+
+The following code demonstrates how to use this provider:
+```csharp
+var basicAuthProvider = new BasicAuthenticationProvider(() =>
+{
+    // JIRA API expects credentials in the format "email:apikey"
+    return Task.FromResult(
+        Env.Var("MY_EMAIL_ADDRESS") + ":" + Env.Var("JIRA_API_KEY")
+    );
+});
+var skill = kernel.ImportOpenApiSkillFromResource(SkillResourceNames.Jira, basicAuthProvider.AuthenticateRequestAsync);
+```
+
+### [`BearerAuthenticationProvider`](./BearerAuthenticationProvider.cs)
```

Comment:
Would be helpful to point to the Example22 in kernel-syntax-examples here

Reply:
Let's hold off until yours & Aman's PRs go in, since the filename(s) for the OpenAPI skill examples are changing so any links I add will get broken. I'll send another PR out to add the reference here once those changes go in.